File: BIO_2017_stage1/1a.py
import sys
first_row = sys.stdin.readline()

first_row = first_row.strip()

def compute_color(A, B):
    if A == B:
        return A
    if B < A:
        C = A
        A = B
        B = C
    if A == "B" and B == "R":
        return "G"
    elif A == "G" and B == "R":
        return "B"
    elif A == "B" and B == "G":
        return "R"

current_row = first_row
#print(current_row)
for j in range(len(current_row) - 1):
    next_row = []
    for i in range(len(current_row) - 1):
        letter = compute_color(current_row[i], current_row[i + 1])
        next_row.append(letter)
    current_row = "".join(next_row)
#    print(current_row)
print(current_row)

File: BIO_2017_stage1/1a.vb
Module Triangles

  Function ComputeColor(Byref A as Char, Byref B as Char) as Char
    If A = B
      return A
    End If
    If B < A:
      Dim C as String
      C = A
      A = B
      B = C
    End If
    If A = "B" and B = "R"
      return "G"
    Else If A = "G" and B = "R"
      return "B"
    Else If A = "B" and B = "G"
      return "R"
    End If
  End Function

  Function Main() as Integer

    Dim FirstRow as String = Console.ReadLine()
    Dim Index as Integer
    Dim OuterIndex as Integer
    Dim CurrentRow() As Char
    Dim NextRow() As Char
    Dim FirstRowSize As Integer = Len(FirstRow)

    CurrentRow = New Char(FirstRowSize) {}

    For Index = 0 To FirstRowSize - 1
      CurrentRow(Index) = FirstRow(Index)
    Next
    For OuterIndex = 0 To (FirstRowSize-2)
      NextRow = New Char(Len(CurrentRow)-1) {}
      For Index = 0 To (Len(CurrentRow)-2)
        Dim v as Char = ComputeColor(CurrentRow(Index), CurrentRow(Index + 1))
        NextRow(Index) = v
      Next
      CurrentRow = NextRow
    Next
    Console.WriteLine(NextRow(0))
    return 0
  End Function

End Module
